### PR TITLE
[Term Entry] Go String Functions Index()

### PR DESCRIPTION
**Pull request by Manikanta528**
### Description

Addressing issue #2564 

### Type of Change

- Adding a new term entry  Go String Functions Index()

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/strings/terms/index/index.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+```pseudo
+strings.Index(str, sub_str)
+```
+
+Where `str` is the original string and `sub_str` is the substring to find the first occurrence in the original string.
+
+## Example
+
+The following example demonstrates the use of the `strings.Index()` function.
```
Comment (dakshdeepHERE) [inline content/go/concepts/strings/terms/index/index.md:27]:
```suggestion
The following example demonstrates the use of the `strings.Index()` function:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
1 change

File: content/go/concepts/strings/terms/index/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+The output will be:
+
+```shell
+2
+-1
+```
+
+## Codebyte Example
```
Comment (caupolicandiaz) [inline content/go/concepts/strings/terms/index/index.md:48]: I would recommend removing the codebyte unless you can revise the example to demonstrate some other aspect or functionality of the method that isn't demonstrated by the other example.

**Review by caupolicandiaz — APPROVED** [review]
@Manikanta528 Overall, looks good. I just left one suggestion. This is ready for a second maintainer review.

File: content/go/concepts/strings/terms/index/index.md (hunk 2 above)
Comment (Manikanta528) [reply content/go/concepts/strings/terms/index/index.md:48]: Okay

**Review by SSwiniarski — APPROVED** [review]
@Manikanta528 looks good to go.

**Comment by SSwiniarski** [comment]
@Manikanta528 congrats on the new entry! Live link here: https://www.codecademy.com/resources/docs/go/strings/index

image

**Pull request merged**